python/sanlock.c

```diff
@@ -81,6 +81,17 @@ is_py_integer(PyObject *obj)
 #endif
 }
 
+/* Note that we must keep the input object alive */
```

Comment:
> We should also make sure that we actually return NULL and raise TypeError in both py2 and py3.

This is the documented behavior in case of failure:
py2: https://docs.python.org/2/c-api/string.html#c.PyString_AsString
py3: https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_AsUTF8AndSize (PyUnicode_AsUTF8 doc refers to this)

Reply:
Since you got the links, lets include them in the comment.